R/progeny.r

```diff
@@ -1,64 +1,204 @@
+#'                 than 1. A logical value. TRUE: the null distributions
+#'                 generated to assess the signifance of the pathways scores 
+#'                 is also returned. 
+#' @param assay_name Only applys if the input is a Seurat object. It select the
+#'               name of the assay on which Progeny will be run. Default to: 
+#'               RNA, i.e. normalized expression values.
+#' @param return_assay Only applys if the input is a Seurat object. A logical 
+#'               value indicating whether to return progeny results as a new 
+#'               assay called Progeny in the Seurat object used as input. 
+#'               Default to FALSE. 
+#' @param ...    Additional arguments to be passed to the functions. 
+#'               
+#' @return       A matrix with samples in columns and pathways in rows. In case
+#'               we run the method with permutations and the option get_nulldist
+#'               to TRUE, we will get a list with two elements. The first 
+#'               element is the matrix with the pathway activity as before. 
+#'               The second elements is the null distributions that we generate
+#'               to assess the signifance of the pathways scores. 
 #' @export
+#' @seealso \code{\link{progenyPerm}}
 #' @examples
-#' # use your gene expression matrix here, this is just for illustration
-#' gene_expression = matrix(rep(1, nrow(model)),
-#'     dimnames=list(rownames(model), "sample"))
+#' # use example gene expression matrix here, this is just for illustration
+#' gene_expression <- as.matrix(read.csv(system.file("extdata", 
+#' "human_input.csv", package = "progeny"), row.names = 1))
 #'
 #' # calculate pathway activities
-#' pathways = progeny(gene_expression)
-progeny = function(expr, scale=TRUE) {
+#' pathways <- progeny(gene_expression, scale=TRUE, 
+#'     organism="Human", top = 100, perm = 1)
+progeny = function(expr, scale=TRUE, organism="Human", top = 100, perm = 1, 
+    verbose = FALSE, z_scores = FALSE, get_nulldist = FALSE, assay_name = "RNA", 
+    return_assay = FALSE, ...) {
     UseMethod("progeny")
 }
 
 #' @export
-progeny.ExpressionSet = function(expr, scale=TRUE) {
-    progeny(Biobase::exprs(expr), scale=scale)
+progeny.ExpressionSet = function(expr, scale=TRUE, organism="Human", top = 100,
+    perm = 1, verbose = FALSE, z_scores = FALSE, get_nulldist = FALSE, ...) {
+    
+    progeny(Biobase::exprs(expr), scale=scale, organism=organism, top=top, 
+        perm = perm, verbose = verbose,  z_scores = z_scores, 
+        get_nulldist = get_nulldist)
+}
+
+#' @export
+progeny.Seurat = function(expr, scale=FALSE, organism="Human", top = 100,
+    perm = 1, verbose = FALSE, z_scores = FALSE, get_nulldist = FALSE, 
+    assay_name = "RNA", return_assay = FALSE,...) {
+    
+    requireNamespace("Seurat")
+    
+    if (!is.logical(return_assay)){
+        stop("return_assay should be a logical value")
+    }
+    
+    if (scale & return_assay){
```

Comment:
Here you return a warning when "scale == T" but still you run the progeny function with scale == T, right? I think you should add in the warning that scale is set to False and write in line 111 - scale = F